L1Trigger/TrackFindingTMTT/src/TrackFitGeneric.cc

```diff
@@ -0,0 +1,52 @@
+///=== This is the base class for all the track fit algorithms
+
+///=== Written by: Alexander D. Morton and Sioni Summers
+
+#include "L1Trigger/TrackFindingTMTT/interface/TrackFitGeneric.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Settings.h"
+#include "L1Trigger/TrackFindingTMTT/interface/Utility.h"
+#include "L1Trigger/TrackFindingTMTT/interface/ChiSquared4ParamsApprox.h"
+#include "L1Trigger/TrackFindingTMTT/interface/KFParamsComb.h"
+#include "L1Trigger/TrackFindingTMTT/interface/SimpleLR.h"
+#ifdef USE_HLS
+#include "L1Trigger/TrackFindingTMTT/interface/HLS/KFParamsCombCallHLS.h"
+#endif
+#include "FWCore/Utilities/interface/Exception.h"
+#include <map>
+#include <new>
+
+namespace TMTT {
+
+  //=== Set configuration parameters.
+
+  TrackFitGeneric::TrackFitGeneric(const Settings* settings, const string& fitterName)
+      : settings_(settings), fitterName_(fitterName), nDupStubs_(0) {}
+
+  //=== Fit a track candidate obtained from the Hough Transform.
+  //=== Specify which phi sector and eta region it is in.
+
+  L1fittedTrack TrackFitGeneric::fit(const L1track3D& l1track3D) {
+    return L1fittedTrack(settings_, l1track3D, l1track3D.getStubs(), 0, 0, 0, 0, 0, 0, 0, 999999., 0);
+  }
+
+  TrackFitGeneric* TrackFitGeneric::create(std::string fitter, const Settings* settings) {
+    if (fitter.compare("ChiSquared4ParamsApprox") == 0) {
```

Comment:
```suggestion
    if (fitter == "ChiSquared4ParamsApprox") {
```
also below (assuming this code stays somewhere)

Reply:
done